PR: https://github.com/odoo/odoo/pull/

From: 503b8ed6925eddc36afd05683eaaad1a93312c8d
From: Adrian Torres
Date: 2018-06-11 07:47:34

Structural Changes: 4
Total Changes: 50

[FIX] *: cleanup (un|re)installation hacks

Replace hacks introduced in V11 by the proper solution.

================================= pseudo patch: =================================

--- a/addons/website/models/__init__.py
+++ b/addons/website/models/__init__.py
@@ -10,6 +10,4 @@ from . import ir_ui_view
 from . import res_company
 from . import res_partner
 from . import res_config_settings
-from . import ir_model_fields
-from . import ir_model
 from . import res_users

--- a/addons/website/models/ir_model.py
+++ b/None
@@ -1,15 +0,0 @@
-# -*- coding: utf-8 -*-
-# Part of Odoo. See LICENSE file for full copyright and licensing details
-
-from odoo import api, models
-
-
-# This is a nasty hack, targeted for V11 only
-class IrModel(models.Model):
-    _inherit = 'ir.model'
-
-    @api.multi
-    def unlink(self):
-        self.env.cr.execute(
-            "DELETE FROM ir_model_fields WHERE name='website_id'")
-        return super(IrModel, self).unlink()

--- a/addons/website/models/ir_model_fields.py
+++ b/None
@@ -1,29 +0,0 @@
-# -*- coding: utf-8 -*-
-# Part of Odoo. See LICENSE file for full copyright and licensing details
-
-from odoo import models
-
-
-# This is a nasty hack, targeted for V11 only
-class IrModelFields(models.Model):
-    _inherit = 'ir.model.fields'
-
-    def unlink(self):
-        # Prevent the deletion of some `shared` fields... -_-
-        social = (
-            'social_facebook',
-            'social_github',
-            'social_googleplus',
-            'social_linkedin',
-            'social_twitter',
-            'social_youtube',
-        )
-
-        self = self.filtered(
-            lambda rec: not (
-                (rec.model == 'res.company' and rec.name in social)
-                or (rec.model == 'res.config.settings'
-                    and rec.name == 'auth_signup_uninvited')
-            )
-        )
-        return super(IrModelFields, self).unlink()

--- a/addons/website/models/res_config_settings.py
+++ b/addons/website/models/res_config_settings.py
@@ -13,8 +13,8 @@ class ResConfigSettings(models.TransientModel):
     def _default_website(self):
         return self.env['website'].search([], limit=1)
 
-    # FIXME: Set website_id to ondelete='cascade' in master
-    website_id = fields.Many2one('website', string="website", default=_default_website, required=True)
+    website_id = fields.Many2one('website', string="website",
+                                 default=_default_website, required=True, ondelete='cascade')
     website_name = fields.Char('Website Name', related='website_id.name')
     language_ids = fields.Many2many(related='website_id.language_ids', relation='res.lang')
     language_count = fields.Integer(string='Number of languages', compute='_compute_language_count', readonly=True)
